### PR TITLE
Regtools recipe addition

### PR DESCRIPTION
**Pull request by luke-c-sargent**
* [X] I have read the guidelines above.
* [X] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

[The tool this recipe describes](https://github.com/griffithlab/regtools) is a dependency of a toolshed entry for [Galaxy](https://galaxyproject.org/) found [here](https://toolshed.g2.bx.psu.edu/view/yating-l/regtools_junctions_extract/01ed8e112f2a).

As Galaxy is moving towards conda for all its dependency resolution / package management, I am creating recipes for key components of my group's [particular Galaxy workflow](http://gonramp.wustl.edu/).

This is my first conda recipe, so per the docs some review is in order -- though I can't seem to ping @bioconda/core, so instead I will ping the two people who are in both bioconda and galaxy people pages: @shiltemann @yhoogstrate 

([travis output](https://travis-ci.org/luke-c-sargent/bioconda-recipes))

### CODE REVIEW

File: recipes/regtools/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  run:
+  - libgcc # [not osx]
+about:
+  home: https://github.com/griffithlab/regtools
+  license: MIT
+  summary: "Tools that integrate DNA-seq and RNA-seq data to help interpret mutations in a regulatory and splicing context."
+
+test:
+  commands:
+  - which regtools
```
Comment (bgruening) [inline recipes/regtools/meta.yaml]: Do you have a better test for this tool? Is `regtools --help` or `--version` working?

File: recipes/regtools/build.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+mkdir build
+cd build/
+cmake ..
+make
+cp regtools $PREFIX/bin
+chmod +x $PREFIX/bin/regtools
```
Comment (bgruening) [inline recipes/regtools/build.sh]: This chmod is not needed afaik.

File: recipes/regtools/meta.yaml (hunk 1 above)
Comment (luke-c-sargent) [inline recipes/regtools/meta.yaml]: i will make the default output into a test, though i have a question regarding that. 

the tool's 0.3.0 release erroneously says '0.2.0' in its --help output. this is fixed in a non-release commit (several commits after release), but is confusing to anyone who would run --help. i assume i should stick with the release version, but should I make a note of the inconsistency somewhere?

File: recipes/regtools/run_test.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+#help output is to stderr
+HELP="$(regtools --help 2>&1 >/dev/null)"
+VERSION="$(regtools --version 2>&1 >/dev/null)"
```
Comment (bgruening) [inline recipes/regtools/run_test.sh]: Oh that is actually nice. But really not needed. We are happy if --version or --help is returning something that does not look like an error. Sorry. Can you just add --version and --help and remove this file?

**Comment by bgruening** [comment]
@luke-c-sargent its not a timeout the error is here: https://travis-ci.org/bioconda/bioconda-recipes/jobs/230511484#L1685

**Comment by bgruening** [comment]
If zlib is needed please include it, can you provide me a link to a timeout.

**Comment by luke-c-sargent** [comment]
@bgruening you have discovered my struggle!

timed out:
https://travis-ci.org/luke-c-sargent/bioconda-recipes/builds/230482260

it _seems_ as though just calling the executable 'regtools' does not cause the shared library error:
- [when running the original script, --version and --help both create the same output simply executing regtools does](https://github.com/luke-c-sargent/bioconda-recipes/commit/4ae0e360c4f0f041dbbde355d283e1aa128fbf43). no zlib inclusions, passes.
- [when running without zlib included and a command argument, e.g `regtools junctions`](https://travis-ci.org/luke-c-sargent/bioconda-recipes/jobs/230403565#L1680), the shared library error.
- when running with zlib and a command argument to regtools as linked to above, the process takes a long time and seemingly times out.
- [when running with zlib and no command arguments, just `regtools`](https://travis-ci.org/luke-c-sargent/bioconda-recipes/builds/230471004), it passes.

note: regtools with the command arguments output instructive text and have an exit code of 0, as tested in bash:
```
$ regtools junctions; echo "\nexit code:" $?

Program:	regtools
Version:	0.2.0
Usage:		regtools junctions <command> [options]
Command:	extract		Identify exon-exon junctions from alignments.
		annotate	Annotate the junctions.

exit code: 0
```

**Comment by bgruening** [comment]
Can it be on the `regtools` side, that the zlib stuff is only imported if you rally call the program besides --help?
Can you please include zlib and gcc as runtime dependency, exclude libgcc as runtime and test again. Just a guess.

**Comment by bgruening** [comment]
Sorry that you encounter such problem for your first packages, it's not normal :)

**Comment by luke-c-sargent** [comment]
GCC addition throws this error:
```
[May  9 23:01:30] SERR ERROR conda.core.link:_execute_actions(330): An error occurred while installing package 'defaults::gcc-4.8.5-7'.
[May  9 23:01:30] SERR LinkError: post-link script failed for package defaults::gcc-4.8.5-7
[May  9 23:01:30] SERR running your command again with `-v` will provide additional information
[May  9 23:01:30] SERR location of failed script: /usr/local/bin/.gcc-post-link.sh
```
i have tried a few different permutations of requirements:
- **build**: [ gcc/llvm, cmake ] **run** [ gcc, zlib ]
    `LinkError: post-link script failed for package defaults::gcc-4.8.5-7`
- **build**: [ gcc/llvm, cmake ] **run** [ zlib ]
    `regtools: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory`
- **build**: [ gcc/llvm, cmake, zlib ] **run** [ gcc, zlib ]
    `LinkError: post-link script failed for package defaults::gcc-4.8.5-7`
- **build**: [ gcc/llvm, cmake, zlib ] **run** [ libgcc, gcc, zlib ]
    `LinkError: post-link script failed for package defaults::gcc-4.8.5-7`

the link errors also say:
`Task processing failed: Unexpected exit code [1] of container [c35e12c1df0b step-5c7137427e], container preserved`

could this be related to [this issue you commented on in January](https://github.com/bioconda/bioconda-recipes/issues/3496)?

**Comment by bgruening** [comment]
Ok back to the beginning. This is how it should be. I will try to run it locally and figure out why it times out.

**Comment by bgruening** [comment]
Oh great!
```
09:29:30 BIOCONDA DEBUG Tests to run: regtools && regtools junctions
09:29:30 BIOCONDA DEBUG mulled-build command: ['mulled-build', 'build-and-test', 'regtools=0.3.0--0', '-n', 'biocontainers', '--test', 'regtools && regtools junctions', '--extra-channels', 'file://home/bag/miniconda3/conda-bld,r,defaults,conda-forge']
09:30:29 BIOCONDA INFO TEST SUCCESS recipes/regtools, CONDA_BOOST=1.61;CONDA_GMP=5.1;CONDA_GSL=1.16;CONDA_HDF5=1.8.17;CONDA_HTSLIB=1.4;CONDA_NCURSES=5.9;CONDA_NPY=110;CONDA_PERL=5.22.0;CONDA_PY=27;CONDA_R=3.3.1

```

**Comment by luke-c-sargent** [comment]
am i correct in assuming progress is stalled because it [only passes with a modification to bioconda's travis script](https://github.com/luke-c-sargent/bioconda-recipes/commit/a1986f19b69b2ff4ac5c4aa21586d9c9c5f6b4da), and that [using the canonical travis script causes failure](https://travis-ci.org/bioconda/bioconda-recipes/jobs/230657540)?

not sure how to progress from here :/
tips appreciated, sorry for being problematic

**Comment by bgruening** [comment]
I can build it locally but not here, not sure currently what is going on :(

**Comment by johanneskoester** [comment]
We  have made changes to our build system (migrated from Travis CI to  Circle CI). If you would like to continue working on this PR, please  reopen this PR and update with latest upstream/master.

**Pull request closed without merging**